indico/modules/events/registration/client/js/form/fields/TimetableSessions.jsx

```diff
@@ -0,0 +1,257 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import timeTableURL from 'indico-url:event_registration.api_event_timetable';
+
+import PropTypes from 'prop-types';
+import React, {useState, useEffect} from 'react';
+import {useSelector} from 'react-redux';
+import {Checkbox, Form} from 'semantic-ui-react';
+
+import {
+  FinalCheckbox,
+  FinalRadio,
+  FinalField,
+  FinalInput,
+  validators as v,
+} from 'indico/react/forms';
+import {useIndicoAxios} from 'indico/react/hooks';
+import {Translate} from 'indico/react/i18n';
+
+import '../../../styles/regform.module.scss';
+import {getStaticData} from '../selectors';
+
+function TimetableSessionsComponent({
+  value,
+  name,
+  sessionData,
+  onChange,
+  allowFullDays,
+  display,
+  minimum,
+  maximum,
+  ...props
+}) {
+  function handleValueChange(e, data) {
+    const id = data.value;
+    const newValue = value.includes(id) ? value.filter(el => el !== id) : [...value, id];
+    onChange(newValue);
+  }
+
+  const minMsg = minimum > 0 ? `at least ${minimum}` : ``;
+  const maxMsg = maximum > 0 ? `no more than ${maximum}` : ``;
+  const and = minimum > 0 && maximum > 0 ? ` and ` : ``;
+  const validatorMsg =
+    maximum > 0 || minimum > 0 ? `Please select ${maxMsg} ${and} ${minMsg} session blocks. ` : ``;
+  const currentSelectionMsg = `You selected ${value.length} session blocks.`;
```

Comment:
This belongs in the validate prop of the `FinalField`. Have a look at e.g. `MultiChoiceInput` for an example ;)